### ISSUE TITLE
`whenStable` is not waiting for some animations

### ISSUE THREAD

**Issue by juliemr** [angular/protractor#2154]
Particularly, animations from the angular material project such as https://github.com/angular/material/blob/master/src/components/bottomSheet/bottomSheet.js

**Comment by ThomasBurleson** [comment]
See [BottomSheet - $animate.enter()](https://github.com/angular/material/blob/master/src/components/bottomSheet/bottomSheet.js#L166-L173)

**Comment by sjelin** [comment]
Is this a protractor bug or an angular bug?

**Comment by juliemr** [comment]
It's an angular bug. It's a problem with testability's `whenStable`. But it shows up in Protractor, so I think tracking here makes sense.

**Comment by sjelin** [comment]
Should it be labeled `external bug` then?

**Comment by QuantumInformation** [comment]
Any idea then the external bug will be fixed?

**Comment by mgiambalvo** [comment]
This issue dates back to pre-1.0 Angular Material, however it still seems to be a problem. I focused on bottom-sheet.js. I found that testability works as expected for $aninmate.enter(). However, there seems to be some kind of race condition when the element is removed. I found that roughly 70% of the time, `whenStable` returns after `onRemove` is called, but before the `$animate.leave()` promise resolves. I'll file an issue.
